modules/configuration/php/diagnosis.class.inc

```diff
@@ -0,0 +1,285 @@
+        $values['instruments']     = $instruments;
+        $values['sourceFields']    = $sourceFields;
+        $values['diagnosisTracks'] = $diagnosisTracks;
+
+        return $values;
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        $values = $request->getParsedBody();
+
+        $dxEvolutionID  = $values['DxEvolutionID'] ?? null;
+        $name           = $values['Name'] ?? null;
+        $projectID      = $values['ProjectID'] ?? null;
+        $visit          = $values['visitLabel'] ?? null;
+        $instrumentName = $values['instrumentName'] ?? null;
+        $sourceFields   = explode(',', $values['sourceField']) ?? [];
+        $orderNumber    = $values['orderNumber'] ?? null;
+
+        /**
+        * VALIDATE THE FORM
+        */
+        // Validation: Form is complete
+        if (!($dxEvolutionID && $name && $visit
+            && $instrumentName && $sourceFields && $orderNumber)
+        ) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Please fill out all fields!'
+            );
+        }
+
+        // Validation: Visit is part of Project's config
+        $projectVisits = \Utility::getVisitList(new \ProjectID($projectID));
+        if (!array_key_exists($visit, $projectVisits)) {
+            return new \LORIS\Http\Response\JSON\Conflict(
+                'Visit is not defined for selected Project.'
+            );
+        }
+
+        // Validation: Instrument is part of Visit's test battery
+        $visitInstruments = \Utility::getVisitInstruments($visit);
+        if (!array_key_exists($instrumentName, $visitInstruments)) {
+            return new \LORIS\Http\Response\JSON\Conflict(
+                'Instrument does not exist in selected visit.'
+            );
+        }
+
+        // Validation: Source Field belongs to Instrument
+        $instrumentFields = array_column(
+            \Utility::getSourcefields($instrumentName),
+            'SourceField'
+        );
+        $matches          = array_intersect($sourceFields, $instrumentFields);
+        if (count($matches) !== count($sourceFields)) {
+            return new \LORIS\Http\Response\JSON\Conflict(
+                'Source Field does not exist in instrument.'
+            );
+        }
+
+        // Validation: Order Number should be unique for specified project
+        $DB    = \NDB_Factory::singleton()->database();
+        $match = $DB->pselectOne(
+            "SELECT Name
+            FROM diagnosis_evolution
+            WHERE ProjectID=:projID
+            AND orderNumber=:ord",
+            [
+                'projID' => $projectID,
+                'ord'    => $orderNumber
```

Comment:
```suggestion
                'ord'    => $orderNumber,
                'dxEvoID' => $dxEvolutionID
```

this validation returns conflict error for orderNumber if you're updating an existing evolution